node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -478,82 +483,268 @@ mod tests {
         }
     }
 
-    fn unwrap_single<T: Debug>(vec: Vec<T>) -> T {
-        let mut iter = vec.into_iter();
+    struct TestState {
+        rng: TestRng,
+        state: State<TestContext>,
+        instance_id: u64,
+        fd: FinalityDetector<TestContext>,
+        validators: HashMap<ValidatorIndex, ActiveValidator<TestContext>>,
```

Comment:
You could use `ValidatorMap` here; it's more efficient because it's a `Vec` inside.

Reply:
Also, we really need a better name for `ActiveValidator`… anyway, we shouldn't just call this `validators`, because `validator` is already used in `validators.rs` for a struct containing the validator ID and weight.

Maybe `VertexProposer`…? :confused:

Reply:
Renamed to `active_validators` and changed to use `ValidatorMap` in [`3e59e5b` (#602)](https://github.com/CasperLabs/casper-node/pull/602/commits/3e59e5be187011e0da8f7801704d291ba64967fc).